core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -467,13 +474,42 @@ class KafkaZkClient private[zk] (zooKeeperClient: ZooKeeperClient, isSecure: Boo
    * @param rootEntityType entity type
    * @param sanitizedEntityName entity name
    * @throws KeeperException if there is an error while setting or creating the znode
+   * @throws ControllerMovedException if no controller is defined, or a KRaft controller is defined
    */
   def setOrCreateEntityConfigs(rootEntityType: String, sanitizedEntityName: String, config: Properties): Unit = {
+    val controllerZkVersion = if (!enableEntityConfigNoController) {
```

Comment:
out of curiosity. Why we have a flag to control the checks even though there is a true issue?

Reply:
We still need to support `kafka-config.sh --zookeeper` until 4.0. Previously, it was possible to set configs with the tools even if the cluster wasn't running. This flag lets us configure the whole KafkaZkClient in CommandConfig to allow this behavior. Otherwise, it's disabled.